specs/waku/v2/waku-filter.md

```diff
@@ -120,9 +146,24 @@ messages to the node. This period is up to the consumer of the protocol and node
 implementation, though a reasonable default is one minute.
 
 ---
+# Security Analysis 
+
+- **Prevention of Denial of Service**: DoS attack can be mitigated through accounting model as provided by [Waku Swap Accounting specs](https://github.com/vacp2p/specs/blob/master/specs/waku/v2/waku-swap-accounting.md). In a nutshell, nodes have to pay for the service they obtain from each other, which means, in terms of the `WakuFilter` protocol, the subscribing node will be charged for the messages that it obtains from other full nodes. In addition to incentivizing the service provider, accounting also makes DoS attacks costly for malicious nodes.
```

Comment:
Doesn't this mean that this protocol on its own doesn't support it? But you have to use something else (swap) to get it?

Reply:
You are certainly right, however, my initial thought was that in the security analysis section, we are giving directions to users on how to achieve certain security features by all the protocols provided in waku. Given that the  swap accounting protocol is part of waku v2 and enables DoS protection,  so does the filter protocol by deploying the swap accounting protocol.

I will remove the DoS from the filter and leave this feature to be discussed in the accounting specs.

Reply:
It can be mentioned but more as a "Note on usage" IMO. It is not the primary design goal of filter as a protocol.